themes/default/content/what-is/what-is-infrastructure-as-code.md

```diff
@@ -62,85 +62,105 @@ video:
+
+### Frequent Infrastructure Changes
 
-The second trend is cloud modernization. After organizations migrate to the cloud, they tend to look for opportunities to maximize the value they get from their cloud environment. This frequently involves adopting technologies such as serverless, containers and Kubernetes, which, when applied correctly, these technologies enable teams to deliver value more quickly, and uses managed services to offload some of the heavy lifting to the cloud provider. These technologies and services generally require a more granular management of infrastructure. Stitching together all the primitives that the cloud provider offers into solutions that serve the business is a great fit for infrastructure as code.
+Finally, the rate of change for a company's infrastructure is increasing. Part of this increase in the rate of change is due to cloud adoption and cloud modernization. There is a third reason, though: organizations are finding that they can move faster if they take advantage of the fundamental elasticity of the cloud.
 
-### Frequent infrastructure changes
+For teams managing tens or hundreds of cloud resources that change once every few months, managing infrastructure using scripts or via interactive means (such as using a UI or a CLI) might still be possible. More commonly, teams finding themselves managing thousands or tens of thousands of resources that change daily or even hourly. Embracing automation via infrastructure as code is the only way to take control of that kind of complexity.
 
-Finally, the rate at which a company’s infrastructure changes is increasing. Cloud adoption and cloud modernization are two of the reasons this is happening. The third is that organizations are finding that they can move faster if they take advantage of the fundamental elasticity of the cloud.
+## What are the Key Elements of Infrastructure as Code?
 
-For teams managing tens or hundreds of cloud resources that change once every few months, it may still be possible to manage infrastructure with point-and-click or scripts. But the more common situation today is that teams are managing thousands or tens of thousands of resources that change daily or even hourly. Infrastructure as code is how you take control of that kind of complexity.
+The key elements of infrastructure as code are the same key elements you'd find in the majority of software engineering environments. These include:
+
+1. **An infrastructure as code tool:** Infrastructure as code tools may be provided by and limited to a single cloud provider (AWS Cloudformation is one example), or may support multiple cloud providers. Tools may be limited to supporting YAML or JSON; may require the use of a specialized and proprietary domain-specific language (DSL); or may support the use of general purpose programming languages such as TypeScript/JavaScript, C#, Go, Python, and Java.
```

Comment:
its a bit tautological to define elements of IaC with IaC tool right?

Should this instead be like programming model? or resource model?

Reply:
I have mixed feelings on this. On one hand, we define IaC (in the first paragraph) as "bringing software engineering principles, approaches, and tools into the cloud infrastructure space", which means that including an IaC tool is appropriate. OTOH, your comment has merit. However, I don't feel that "programming model" or "resource model" are appropriate here. In reality, you can't do IaC without an IaC tool. Perhaps some rewording would be appropriate here.

Reply:
maybe IaC engine makes it sound a bit more meaty?

Reply:
I've made some changes here that I think address the concern, but your feedback is welcome.